tests/docs/test_markup.py

```
@@ -0,0 +1,201 @@
+])
+
+_ONE_COLON = re.compile(r'^\s*\.\.\s+([a-z][a-z0-9-]*):(?!:)')
+
+# A strong span, by the reStructuredText inline-markup rules: the opening ** is not
+# followed by whitespace, the closing ** is not preceded by whitespace, and the span
+# holds no ** of its own. Nested markup is a backtick or a lone asterisk inside it.
+_STRONG = re.compile(r'\*\*(?![\s*])((?:(?!\*\*).)+?)(?<![\s*])\*\*')
+_NESTED = re.compile(r'`|(?<![\w*])\*(?![\s*])')
+
+
+def _uncode(lines):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_uncode`

(ANN202)